### PR TITLE
guard against null _disabledEventEmissions in Observer#fireEvent

### LINKED ISSUE #1975: Cannot read property 'includes' of undefined
Uncaught TypeError: Cannot read property 'includes' of undefined
    at e.fireEvent (observer.js:125)
    at HTMLUnknownElement.<anonymous> (region.js:349)

### Wavesurfer.js version(s):
https://unpkg.com/wavesurfer.js
https://unpkg.com/wavesurfer.js/dist/plugin/wavesurfer.regions.min.js

### Browser and operating system version(s):
Chrome 83, Windows 10

### Code needed to reproduce the issue:
```
<script src="https://unpkg.com/wavesurfer.js/dist/plugin/wavesurfer.regions.min.js"></script>

// example region
wavesurfer.addRegion({
		id: "pad-1",
		start: 1,
		end: 10,
		color: "blue",
});
```
### Use behaviour needed to reproduce the issue:
Error is produced on interactions with the wavesurfer regions UI
ex. hovering over a region

*edit - the initialized regions appear to load correctly.. but attempting to clearRegions() and load new produces the error above, and the new regions fail.

### PR DESCRIPTION
**Pull request by osheroff**
due to the odd way the RegionsPlugin pulls in the methods in util.Observer, it needs to setup a private array or it crashes in `fireEvent`.

I could also fix this in `Observer` by checking for the existence of the array.  I dunno.  both seem odd.

fixes #1975

### CODE REVIEW

**Comment by osheroff** [comment]
hmmm... this doesn't seem to be fixing the crash

**Comment by osheroff** [comment]
nevermind.  I'm doing something dumb.

**Comment by osheroff** [comment]
@katspaugh I was doing something dumb in my own code *and* in this fix.  But it does need to be fixed, regions are totally broken in master.

File: src/plugin/regions/index.js [hunk 1]
```diff
@@ -123,6 +123,9 @@ export default class RegionsPlugin {
         observerPrototypeKeys.forEach(key => {
             Region.prototype[key] = this.util.Observer.prototype[key];
         });
+
+        Region.prototype["_disabledEventEmissions"] = [];
```
Comment (katspaugh) [inline src/plugin/regions/index.js:127]: Thanks for the quick fix! Could you change the quotes to single?

**Comment by cadavre** [comment]
The problem is with https://github.com/katspaugh/wavesurfer.js/blob/master/src/util/observer.js#L124

Inside `fireEvent` – `this` is being an argument passed when triggering the event, not Observer itself.

It looks like Observer is somehow broken, not Regions plugin.

File: src/plugin/regions/index.js (hunk 1 above)
Comment (cadavre) [reply src/plugin/regions/index.js:127]: Why set `disabledEventEmissions` on region? It should be set on wavesurfer itself only. This is not a fix.

Comment (osheroff) [reply src/plugin/regions/index.js:127]: no, this is the fix.  I had previously opened a PR where it wasn't the fix.

**Comment by osheroff** [comment]
@cadavre test it yourself before and after this patch.  this does indeed fix it...

**Comment by osheroff** [comment]
@katspaugh updated with single quotes

File: src/plugin/regions/index.js (hunk 1 above)
Comment (osheroff) [reply src/plugin/regions/index.js]: @cadavre the code is confusing, but my fix is correct.  On line 123 we grab all the methods from `utils.Observer` and stuff them into `Region`.  but `Observer` now requires that `this` contains an array called `_disabledEventEmissions`.   If you read further you'll see that many objects in the wavesurfer codebase inherit from `Observer`, not just wavesurfer.

Comment (cadavre) [reply src/plugin/regions/index.js]: @osheroff I see now – Plugins are Observers as well (`Plugin extends Observer`).

The `_disabledEventEmissions` was missing, because it's initialized in Observer's constructor. Shouldn't we call super() somewhere while initializing plugins @katspaugh ?

Comment (cadavre) [reply src/plugin/regions/index.js]: Thanks for clarification @osheroff !

Comment (osheroff) [reply src/plugin/regions/index.js]: > shouldn't we call super() somewhere while initializing plugins 

The Region plugin doesn't literally inherit from `Observer`, it kinda just borrows all the code.  I'm not sure why this was done, I think probably because they're built into different source files?  I agree it's kind of goofy what the Region plugin is doing here.

**Comment by thijstriemstra** [comment]
All of the plugins that use observer a broken currently, e.g. microphone plugin. I don't like adding this change to every plugin (unless there's really no other way).

**Comment by osheroff** [comment]
well @thijstriemstra here's the other approach...

File: src/plugin/regions/index.js [hunk 2]
```diff
@@ -123,6 +123,7 @@ export default class RegionsPlugin {
         observerPrototypeKeys.forEach(key => {
             Region.prototype[key] = this.util.Observer.prototype[key];
         });
+
```
Comment (thijstriemstra) [inline src/plugin/regions/index.js:126]: can you revert this whitespace change?

File: src/util/observer.js [hunk 3]
```diff
@@ -115,14 +115,23 @@ export default class Observer {
         this._disabledEventEmissions = eventNames;
     }
 
+    /**
+     * looks overly paranoid but plugins borrow part of this class
+     * without calling the constructor.
+     */
```
Comment (thijstriemstra) [inline src/util/observer.js:121]: can you rephrase this? :)

**Comment by osheroff** [comment]
@thijstriemstra updated
